web/client/components/share/SharePanel.jsx

```diff
@@ -97,27 +105,51 @@ class SharePanel extends React.Component {
             embed: 3
         };
         const bbox = join(this.props.bbox, ',');
+        const coordinate = this.getCoordinates(this.props);
         this.setState({
             bbox,
-            eventKey: tabs[this.props.selectedTab] || 1
+            eventKey: tabs[this.props.selectedTab] || 1,
+            zoom: this.props.zoom,
+            coordinate
         });
     }
 
     UNSAFE_componentWillReceiveProps(newProps) {
-        const bbox = join(this.props.bbox, ',');
         const newBbox = join(newProps.bbox, ',');
-        if (bbox !== newBbox) {
+        if (this.props !== newProps) {
+            const coordinate = this.getCoordinates(newProps);
             this.setState({
-                bbox: newBbox
+                bbox: newBbox,
+                zoom: newProps.zoom,
+                coordinate
             });
         }
     }
 
+
+    getLatLng = (point) =>{
+        const latlng = point && point.latlng || null;
+        let lngCorrected = null;
+        if (latlng) {
+            lngCorrected = latlng && Math.round(latlng.lng * 100000000000000000) / 100000000000000000;
+            lngCorrected = lngCorrected - 360 * Math.floor(lngCorrected / 360 + 0.5);
+        }
+        return { lat: latlng && latlng.lat, lon: lngCorrected};
+    }
+
     getShareUrl = () => {
         const { settings, advancedSettings } = this.props;
         let shareUrl = getSharedGeostoryUrl(removeQueryFromUrl(this.props.shareUrl));
         if (settings.bboxEnabled && advancedSettings && advancedSettings.bbox && this.state.bbox) shareUrl = `${shareUrl}?bbox=${this.state.bbox}`;
         if (settings.showHome && advancedSettings && advancedSettings.homeButton) shareUrl = `${shareUrl}?showHome=true`;
+        if (settings.centerAndZoomEnabled && advancedSettings && advancedSettings.centerAndZoom) {
+            const centerOrMarker = reverse(values(this.state.coordinate)).join(',');
+            if (settings.markerEnabled) {
+                shareUrl = `${shareUrl}?marker=${centerOrMarker}&zoom=${this.state.zoom}`;
+            } else {
+                shareUrl = `${shareUrl}?center=${centerOrMarker}&zoom=${this.state.zoom}`;
+            }
```

Comment:
i think this else branch is not needed
```suggestion
            }
            shareUrl = `${shareUrl}?center=${centerOrMarker}&zoom=${this.state.zoom}`;
```